src/advanced/service-testing.md

```diff
@@ -1,155 +1,260 @@
-# Sandbox Testing
+# Service Testing
 
-Sandbox is a package that simulates the rest of the network for a node.
-The sandbox is used to test [services](../architecture/services.md),
-[the consensus algorithm](consensus/specification.md),
-and [service endpoints](../glossary.md#service-endpoint).
+You can test Exonum services with the help of the [**exonum-testkit**][exonum-testkit]
+crate. The crate allows to test transaction execution and APIs in the synchronous
+environment (that is, without consensus algorithm and network operation involved).
+Tests are executed in the same system process as the service code itself, allowing
+to more easily debug service business logic (for example, panics).
 
-Using sandbox one can send a message to the node and expect some response from it
-which is then checked against the reference response. The sandbox can be used to
-test the consensus algorithm (by sending consensus messages to the node and
-verifying its response) and the operation of the services (by committing
-transactions to the blockchain and verifying the service response). Similarly,
-tests of the public REST API can be performed.
+## Installation
 
-## Sandbox Interface
+In most cases you want to install `exonum-testkit` as a [development dependency][dev-dep],
+so it would be compiled for tests only. For this, add the dependency to
+to the `dev-dependencies` section of the `Cargo.toml` file of your project:
 
-A sandbox instance can be created with `sandbox_with_services` method, which
-allows to specify a list of services for testing.
-
-Sandbox provides network emulation capabilities:
-
-- `recv`  
-  Simulates receiving a message by the node.
-
-- `send`  
-  Checks if a message has been sent by the node.
-
-- `broadcast`  
-  Checks if the node has broadcasted the message of a particular content
-  (e.g. a transaction).
-
-Sandbox also allows to emulate block acceptance and time:
+```toml
+[dev-dependencies]
+exonum-testkit = "0.1"
```

Comment:
`0.1.0` is better, for example, `Racer` ignores crates without full version.

Reply:
Didn't know that, thanks for info!